### PR TITLE
Collect notable improvements of the 0.1.58 release

### PR DESCRIPTION
**Pull request by matejak**

### CODE REVIEW

File: _posts/2021-09-31-release-0.1.58.md [hunk 1]
```diff
@@ -35,7 +35,13 @@ If you would like to bring your CCEs upstream, you will just need to extend the
...
 
 
-### Developer environment setup via Ansible
+### Development environment setup via Ansible
+
+We now have an Ansible role to help us to very quickly prepare a completely functional development environment. About 10 minutes are enough for Ansible to safely conclude the hard work and deliver a ready to go development environment.
+
+This is a great option to start contributing faster, aligned with the official documentation and enjoying the automation benefits. : )
+
+More information is found in the [development documentation](https://complianceascode.readthedocs.io/en/latest/manual/developer/02_building_complianceascode.html).
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:61]:
```suggestion
### Development environment setup via Ansible

We now have an Ansible role to help us to very quickly prepare a completely functional development environment. About 10 minutes are enough for Ansible to safely conclude the hard work and deliver a ready to go development environment.

This is a great option to start contributing faster, aligned with the official documentation and enjoying the automation benefits. : )

More information is found in the [development documentation](https://complianceascode.readthedocs.io/en/latest/manual/developer/02_building_complianceascode.html).
```

Couple of suggestions

File: _posts/2021-09-31-release-0.1.58.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+### Sorting of references and identifiers
+
+### Migration of the shared Bash remediations functions to Jinja macros
+
+Writing remediations is an important part of developing security content.
+And although each rule addresses a single configuration change, different rules may need to run similar commands to remediate the system.
+So it is only natural that the project provides a library to streamline development and reduce code duplication.
+
+The project features two mechanisms that make Bash code available for use in remediations.
+The first mechanism are the [bash_remediation_functions](https://github.com/ComplianceAsCode/content/tree/v0.1.57/shared/bash_remediation_functions).
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:44]:
```suggestion
The first mechanism is the [bash_remediation_functions](https://github.com/ComplianceAsCode/content/tree/v0.1.57/shared/bash_remediation_functions).
```

File: _posts/2021-09-31-release-0.1.58.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+A mechanism that makes use of the XCCDF substitution capability to inject the Bash function into the remediations, this means that OpenSCAP is used to make the substituions and render the final remediation snippet.
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:45]:
```suggestion
A mechanism that makes use of the XCCDF substitution capability to inject the Bash function into the remediations, this means that OpenSCAP is used to make the substitutions and render the final remediation snippet.
```

File: _posts/2021-09-31-release-0.1.58.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+This approach also requires the remediation to declare an include for the build system to transform the Bash function calls into the XCCDF substitution syntax.
+
+The second mechanism are the [Jinja macros](https://github.com/ComplianceAsCode/content/blob/v0.1.57/shared/macros-bash.jinja), which are expanded during the content build.
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:48]:
```suggestion
The second mechanism is the [Jinja macros](https://github.com/ComplianceAsCode/content/blob/v0.1.57/shared/macros-bash.jinja), which are expanded during the content build.
```

File: _posts/2021-09-31-release-0.1.58.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+In this release a good portion of the Bash remediation functions were transformed into Jinja macros, and by next release all of them will have been moved up to macros.
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:50]:
```suggestion
In this release a good portion of the Bash remediation functions were transformed into Jinja macros, and by the next release all of them will be turned into macros.
```

File: _posts/2021-09-31-release-0.1.58.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+This will leave the project with a single mechanism, which should streamline how we define and make available Bash snippets to remediations.
+The expansion of the snippets at build time will provide more clarity on the final remediation;
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:52]:
```suggestion
The expansion of snippets at build time will provide more clarity on the final remediation;
```

File: _posts/2021-09-31-release-0.1.58.md (hunk 1 above)
Comment (yuumasato) [reply _posts/2021-09-31-release-0.1.58.md:61]: @ggbecker It is not clear what the suggestion is changing.

File: _posts/2021-09-31-release-0.1.58.md (hunk 5 above)
Comment (yuumasato) [reply _posts/2021-09-31-release-0.1.58.md]: @ggbecker I think the use of future perfect is correct.

**Review by yuumasato — COMMENTED** [review]
Pushed ac571d4 with a few fixes.

File: _posts/2021-09-31-release-0.1.58.md (hunk 5 above)
Comment (yuumasato) [reply _posts/2021-09-31-release-0.1.58.md]: It is actually passive future perfect. https://www.englishreservoir.com/all-passive-forms/future-perfect-3/

File: _posts/2021-09-31-release-0.1.58.md (hunk 1 above)
Comment (ggbecker) [reply _posts/2021-09-31-release-0.1.58.md:61]: All right, I had some changes here but then I switched to a different page and I lost my suggestions... let me redo it :(

File: _posts/2021-09-31-release-0.1.58.md (hunk 5 above)
Comment (ggbecker) [reply _posts/2021-09-31-release-0.1.58.md]: got it ;)

File: _posts/2021-09-31-release-0.1.58.md (hunk 1 above)
Comment (marcusburghardt) [reply _posts/2021-09-31-release-0.1.58.md:61]: Thanks for the fine adjusts @ggbecker

File: _posts/2021-09-31-release-0.1.58.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+categories: template
+author: Gabriel Becker, Marcus Burghardt, Watson Sato, Matěj Týč
+---
+
+With a bit more than one thousand commits since last release we are happy to announce that ComplianceAsCode / Content v0.1.58 was [released](https://github.com/ComplianceAsCode/content/releases/tag/v0.1.58) on September 24th, 2021.
+
+## Highlight of this release
+
+The project now supports another checking engine: [Script Checking Engine (SCE)](https://www.open-scap.org/features/other-standards/sce/).
+By default, the content is built without SCE checks, check [Building SCE](https://complianceascode.readthedocs.io/en/latest/manual/developer/02_building_complianceascode.html?highlight=SCE#building-sce-non-compliant-content) section in the docs, for
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:13]:
```suggestion
By default, the content is built without SCE checks, check [Building SCE](https://complianceascode.readthedocs.io/en/latest/manual/developer/02_building_complianceascode.html#building-sce-non-compliant-content) section in the docs, for
```

File: _posts/2021-09-31-release-0.1.58.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+how to enable it.
+Note that content built with SCE checks are not SCAP compliant.
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:15]:
```suggestion
Note that content built with SCE checks is not SCAP compliant.
```

File: _posts/2021-09-31-release-0.1.58.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+## What's new in the content?
+
+### New Profiles
+
+Notable new profiles in this release:
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:21]:
```suggestion
Notable new profiles added in this release:
```

File: _posts/2021-09-31-release-0.1.58.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+$ PYTHONPATH=. utils/fix_rules.py  --product products/rhel7/product.yml add-cce accounts_tmout 
+``` 
+
+If you try the code above, you will find out that nothing has changed - the utility is able to detect that the rule already has a RHEL7 CCE.
+
+If you are reading it as a non-RHEL developer, the tool accepts a `--cce-pool` options, that allow introduction of another CCE files than the one in `shared/references/cce-redhat-avail.txt`.
+If you would like to bring your CCEs upstream, you will just need to extend the (cce.py)[https://github.com/ComplianceAsCode/content/blob/master/ssg/cce.py] to integrate it with the script.
+
+
+### Sorting of references and identifiers
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:54]: We are still missing this section text, should we drop it? @matejak @yuumasato

File: _posts/2021-09-31-release-0.1.58.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+In this release a good portion of the Bash remediation functions were migrated to Jinja macros, and by the next release all of them will have been migrated.
+This will leave the project with a single mechanism, which should streamline how we define and make available Bash snippets to remediations.
+The expansion of snippets at build time will provide more clarity on the final remediation;
+And the Jinja templating engine provides a formidable customization power, allowing the Bash snippet to be expanded differently for each product, for example.
+
+### Development environment setup via Ansible
+
+We have now an Ansible [role](https://galaxy.ansible.com/marcusburghardt/ansible_role_openscap) to help us to very quickly prepare a completely functional development environment. About 10 minutes are enough for Ansible to safely conclude the hard work and deliver a ready to go development environment.
+
+This is a great option to start contributing faster, aligned with the official documentation while enjoying the automation benefits. : )
```
Comment (ggbecker) [inline _posts/2021-09-31-release-0.1.58.md:78]:
```suggestion
This is a great option to start contributing faster, aligned with the official documentation while enjoying the automation benefits.
```

**Review by ggbecker — CHANGES_REQUESTED** [review]
we also need to change the name of the file to something else than 31 september (which doesn't exist) for example: 2021-09-3**0**-release-0.1.58.md

File: _posts/2021-09-31-release-0.1.58.md (hunk 10 above)
Comment (yuumasato) [reply _posts/2021-09-31-release-0.1.58.md:54]: I can look into it.

Comment (yuumasato) [reply _posts/2021-09-31-release-0.1.58.md]: Added in e01bf5c

Comment (matejak) [reply _posts/2021-09-31-release-0.1.58.md]: I have asked @Mab879 to give it a thought.

File: _posts/2021-09-30-release-0.1.58.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+
+If you try the code above, you will find out that nothing has changed - the utility is able to detect that the rule already has a RHEL7 CCE.
+
+If you are reading it as a non-RHEL developer, the tool accepts a `--cce-pool` options, that allow introduction of another CCE files than the one in `shared/references/cce-redhat-avail.txt`.
+If you would like to bring your CCEs upstream, you will just need to extend the (cce.py)[https://github.com/ComplianceAsCode/content/blob/master/ssg/cce.py] to integrate it with the script.
+
+
+### Sorting of references and identifiers
+
+The rules in the project can reference security policy requirements to indicate that they implement or support them, and be assigned identifiers to uniqueily identify a configuration item within a product.
```
Comment (ggbecker) [inline _posts/2021-09-30-release-0.1.58.md:55]:
```suggestion
The rules in the project can reference security policy requirements to indicate that they implement or support them, and be assigned identifiers to uniquely identify a configuration item within a product.
```

File: _posts/2021-09-31-release-0.1.58.md (hunk 10 above)
Comment (ggbecker) [reply _posts/2021-09-31-release-0.1.58.md]: I'm ok with current text provided by @yuumasato in e01bf5c

**Review by ggbecker — APPROVED** [review]
LGTM, verified locally on jekyll, no issues with rendering.

**Pull request merged**